### ISSUE TITLE
idea for kick strength

### ISSUE THREAD

**Issue by stephenfournier** [stephenfournier/dotastrikers#154]
1. user hits kick button
2. panorama progress slider appears in bottom center, it goes quickly back and forth from green to red.
3. user hits the kick button again. the more the slider is in the red, the harder the kick will be; the more in the green, the weaker it will be. the kick will go in the direction of the cursor.

harder kicks also make the ball go up in the air more.

**Comment by CarpeSwag** [comment]
wouldn't be feasible unless we had dedicated servers due to lag. Would be useable as a hero's spell b/c you'd just pick anything else, but not as the base mechanic to the game.

**Comment by stephenfournier** [comment]
here is what I'm thinking, this method only requires one server response like the current way we do ball kick. I can't foresee any problems:

1. user clicks ability (can be detected clientside using valve's panorama ability bar)
2. if user is ball controller, do panorama progress slider (clientside). else display 'no-ball error msg'.
3. detect next ability click (clientside)
4. send over pID, cursor pos and strength-number to lua
5. ensure that pID is ball controller again.
6. if pID is ball controller, perform ball kick, else display 'no-ball error msg'.

ball controller's pID can also be sent to panorama on every frame (or probably better if stored in a net-table, i haven't looked much into net-tables yet).

**Comment by stephenfournier** [comment]
```
[09:25 AM]  <+Myll_laptop>	if i need to send something like a pID to panorama on every frame, i should use a net-table right?
[09:27 AM]  <+Perry>	you just shouldn't send stuff to panorama every frame
[09:27 AM]  <+Perry>	if you want panorama to request something every frame net-table seems okay
[09:27 AM]  <+Perry>	although it should only contains information you want everybody to be able to see
[09:28 AM]  <+Myll_laptop>	ok, yeah that's fine
```

**Comment by CarpeSwag** [comment]
The other issue is that this has a huge potential to disrupt the flow of gameplay and possibly even make the games last longer due to the constant disruptions. There won't be quick passes because everything at least requires a tiny bit of delay to not completely get gimped by the slider.

Idunno. If you want to do this drastic of a mechanic I'd suggest opening a thread/poll on the forum about it.

I'm not saying that some sort of new kick mechanic isn't needed, but I don't think this one would work out if we're trying to increase the pace of the game.

**Comment by Soleone** [comment]
any chance it could work like in Fifa where you detect that the button keeps being pressed and only then the progress slider comes up?

* if you press once for less than 100ms or something it would just keep the current behavior.
* if you press longer the progress slider comes up and once you release the key the kick gets executed with current power level

1. not sure if this possible with dota customization
2. keeping a key pressed is not a mechanic players are used to from _any_ custom game probably so it might be a bit unintuitive

but besides those considerations it might be a worthwhile mechanic to introduce if it's possible

**Comment by CarpeSwag** [comment]
I mean, sure, but the point still stands that the mechanic will be flawed without having an option to quick kick or anything along those lines. Maybe just make this slider mechanic like ctrl+w or something or vice versa.

On a diff note, you realize a lot of people who loved the shit out of banjo hate soccer? Even just core Strikers/banjo is enough to make people who dislike the actual sport enjoy the game. Also, people who hate ALL sports won't look at a game called "Strikers" unless they're dragged in by friends. Shifting 100% of our appeal to an audience we barely touch isn't wise nor good design.

**Comment by stephenfournier** [comment]
> but the point still stands that the mechanic will be flawed without having an option to quick kick or anything along those lines. Maybe just make this slider mechanic like ctrl+w or something or vice versa.

You literally just double tap the kick hotkey twice really quickly to make a standard kick (in the direction of your cursor). That is just as fast as the current method in my opinion.

> Also, people who hate ALL sports won't look at a game called "Strikers" unless they're dragged in by friends. Shifting 100% of our appeal to an audience we barely touch isn't wise nor good design.

Games are spread fastest via secondary mediums, like a post about it on the front of reddit, or being dragged by friends (i.e. agario). It was pretty much how Dota Strikers spread so quickly. I initially made a bunch of front-page r/dota2 reddit threads about it. Those guys found the game fun and it spread quickly throughout the whole dota 2 community. Everybody pretty much bypassed the fact that it was a 'sports' game, and simply played it because it was lulzy and fun, not because it was serious. Look at this comment:

![](http://puu.sh/lrbOJ/fdea841e44.png)

**Comment by CarpeSwag** [comment]
If you do, would you mind if I kept the code and split off onto a different mod of the same game?

**Comment by stephenfournier** [comment]
You can absolutely keep and use the code, but you can't use the artwork nor the name 'Dota Strikers'.

**Comment by CarpeSwag** [comment]
The artwork being just the dota strikers logo, correct? If so, that's fine. Thanks.

**Comment by stephenfournier** [comment]
The loading screen and the avatar on the steam workshop.

**Comment by Arhowk** [comment]
Zzz your really gonna split off because you don't like the way kicks work?

In all seriousness both (splitting off and variable kick strength) both
seem like terrible ideas.

**Comment by Soleone** [comment]
yea guys, please don't split off because of a detail like this. i'm sure a middleground can be found and nothing has to be rushed

**Comment by CarpeSwag** [comment]
I agree Arhowk, but (A) I convince garf to help me work on the split off already (which honestly is a huge selling point to working on anything for me), and (B) while I'm not 100% against this change in it's entirety, I know that these clashes are almost definitely going to happen many more times in the future and I'd rather just accept Myll's invitation to go work on something else. I'm probably not going to use most of his code (because rewriting the physics library sounds fun, tbh).

**Comment by stephenfournier** [comment]
Agreed with @CarpeSwag. It's clear we have pretty fundamental differences in how the game should be. I wish him the best of luck and hope he has fun! You're always welcome in the #dota2mods IRC.

That being said - I'm not going to be accepting contributions to Dota Strikers anymore. Thanks everybody for this experience and good luck with your game modes!

**Comment by Arhowk** [comment]
*sigh* does dayne want deh UI upd8s?

**Comment by Soleone** [comment]
don't just think of your guys clashes in opinions also consider the effect on the community when there's now two similar mods. i don't see fragmentation like that benefitting anyone

**Comment by Arhowk** [comment]
Neither do I. One or both will fail. This is evident.

**Comment by stephenfournier** [comment]
Okay, there's competition... so what? There's competition with everything in life.

@Soleone People will choose which game to play. In SC2 there's plenty of similar 'pseudo-sports' custom games. Just like there's different brands of clothing. Big deal.

**Comment by Arhowk** [comment]
Competition in most aspects increases quality of the net system because the
objects are competing, usually for monetary or romantic gains.

In this case, neither set of developers gives a damn so the net quality and
end user satisfaction will decrease as a result of such.

It's inverse evolution.
